lang/rust/avro/src/schema_compatibility.rs

```diff
@@ -430,96 +429,53 @@ impl SchemaCompatibility {
-                            schema_type: String::from("writers_schema"),
+                            schema_type: String::from("readers_schema"),
                             expected_type: vec![SchemaKind::Array],
                         });
                     }
+
+                    return Err(CompatibilityError::TypeExpected {
+                        schema_type: String::from("writers_schema"),
+                        expected_type: vec![SchemaKind::Array],
+                    });
                 }
-                SchemaKind::Date => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::Date,
-                        vec![SchemaKind::Date, SchemaKind::Int],
-                    );
-                }
-                SchemaKind::TimeMillis => {
+                SchemaKind::Date | SchemaKind::TimeMillis => {
                     return check_writer_type(
                         writers_schema,
-                        Schema::TimeMillis,
+                        readers_schema,
                         vec![SchemaKind::Date, SchemaKind::Int],
                     );
                 }
-                SchemaKind::TimeMicros => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::TimeMicros,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
-                    );
-                }
-                SchemaKind::TimestampNanos => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::TimestampNanos,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
-                    );
-                }
-                SchemaKind::TimestampMillis => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::TimestampMillis,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
-                    );
-                }
-                SchemaKind::TimestampMicros => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::TimestampMicros,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
-                    );
-                }
-                SchemaKind::LocalTimestampMillis => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::LocalTimestampMillis,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
-                    );
-                }
-                SchemaKind::LocalTimestampMicros => {
-                    return check_writer_type(
-                        writers_schema,
-                        Schema::LocalTimestampMicros,
-                        vec![SchemaKind::TimeMicros, SchemaKind::Long],
```

Comment:
This is a bug introduced by me few days ago!
It must have been:
`vec![SchemaKind::LocalTimestampMicros, SchemaKind::Long],`
Similar for all above.

Reply:
> Similar for all above

That is, the match could not be grouped.

Reply:
Ah, you meant each type of the match. I get it now. It can be grouped because we can pass r_type: `vec![r_type, SchemaKind::Long]

Reply:
All of them have a `type: long`, so using `r_type` will make the trick and we can grouped all of them :-)